shuup/admin/modules/suppliers/views/delete.py

```diff
@@ -0,0 +1,38 @@
+# This file is part of Shuup.
+#
+# Copyright (c) 2012-2019, Shoop Commerce Ltd. All rights reserved.
+#
+# This source code is licensed under the OSL-3.0 license found in the
+# LICENSE file in the root directory of this source tree.
+from django.conf import settings
+from django.contrib import messages
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import RedirectView
+
+from shuup.core.models import Supplier
+
+
+class SupplierDeleteView(RedirectView):
+    permanent = False
+    pk = None
+
+    def dispatch(self, request, *args, **kwargs):
+        if self.pk is None:
+            self.pk = kwargs.pop("pk")
+        return super(SupplierDeleteView, self).dispatch(request, *args, **kwargs)
+
+    def get_redirect_url(self, *args, **kwargs):
+        if settings.SHUUP_ENABLE_MULTIPLE_SUPPLIERS:
+            return reverse("shuup_admin:shuup_multivendor.vendor.list")
```

Comment:
We can not have this dependency. You have add separate delete view to multivendor addon which implements this view. Could be something like:

```
class SupplierMultivendorDeleteView(SupplierDeleteView):
    def get_redirect_url(self, *args, **kwargs):
            return reverse("shuup_admin:shuup_multivendor.vendor.list")
```